src/gwt/src/org/rstudio/studio/client/projects/Projects.java

```diff
@@ -527,26 +528,46 @@ public void onExecute(final Command continuation)
             {
-               projServer_.createProject(
-                     newProject.getProjectFile(),
-                     newProject.getNewPackageOptions(),
-                     newProject.getNewShinyAppOptions(),
-                     newProject.getProjectTemplateOptions(),
-                     new SimpleRequestCallback<String>()
-                     {
-                        @Override
-                        public void onResponseReceived(String foundProjectFile)
+               Command onReady = () -> {
+                  projServer_.createProject(
+                        newProject.getProjectFile(),
+                        newProject.getNewPackageOptions(),
+                        newProject.getNewShinyAppOptions(),
+                        newProject.getProjectTemplateOptions(),
+                        new SimpleRequestCallback<String>()
                         {
-                           if (!StringUtil.isNullOrEmpty(foundProjectFile))
+                           @Override
+                           public void onResponseReceived(String foundProjectFile)
                            {
-                              // found an existing project file with different name than the
-                              // parent folder; have to update here so that's what we end up
-                              // opening
-                              newProject.setProjectFile(foundProjectFile);
+                              if (!StringUtil.isNullOrEmpty(foundProjectFile))
+                              {
+                                 // found an existing project file with different name than the
+                                 // parent folder; have to update here so that's what we end up
+                                 // opening
+                                 newProject.setProjectFile(foundProjectFile);
+                              }
+                              continuation.execute();
                            }
-                           continuation.execute();
-                        }
-                     });
+                        });
+               };
+               
+               if (newProject.getProjectTemplateOptions() != null)
+               {
+                  // NOTE: We provide built-in project templates for packages that may
+                  // not be currently installed; in those cases, verify that the package is
+                  // installed and if not attempt installation from CRAN first
+                  String pkg = newProject.getProjectTemplateOptions().getDescription().getPackage();
+                  RStudioGinjector.INSTANCE.getDependencyManager().withDependencies(
+                        "Creating project",
+                        "Creating a project with " + pkg,
+                        new Dependency[] { Dependency.cranPackage(pkg) },
+                        false,
+                        (Boolean success) -> { onReady.execute(); });
```

Comment:
Should we execute `onReady` even if we weren't successful?

Reply:
Do we notify the user when package installation fails when invoked through the dependency manager? I recall that in some cases we decided to just fall-through so that if installation failed that at we'd continue execution so the user would see some downstream error (and perhaps ensure any other post-install code that needs to run does run)

Reply:
I don't think so. But if it fails, I think we should display an error here instead of continuing -- e.g. just something like "The package 'foo' is required to create the project, and could not be installed. Try running install.packages('foo') manually.".